internal/gcsx/random_reader.go

```diff
@@ -350,9 +350,8 @@ func (rr *randomReader) ReadAt(
 	// is a 15-20x improvement in throughput: 150-200 MB/s instead of 10 MB/s.
 	if rr.reader != nil && rr.start < offset && offset-rr.start < maxReadSize {
 		bytesToSkip := offset - rr.start
-		p := make([]byte, bytesToSkip)
-		n, _ := io.ReadFull(rr.reader, p)
-		rr.start += int64(n)
+		discardedBytes, _ := io.CopyN(io.Discard, rr.reader, int64(bytesToSkip))
+		rr.start += discardedBytes
```

Comment:
Should we handle error here? Although it is not harmful as we are anyway discarding the reader if start does not meet offset but I think it is good practice to not avoid error handling.

Reply:
+1. Just a log line would help us understand if something goes wrong. otherwise we end up making calls to gcs and never know this failed.

Reply:
Make sense, added a warning log for the error other than io.EOF.